root/layout/sidebar/sidebar-tags.tt

```diff
@@ -1,20 +1,33 @@
-[%- MACRO tag_list(tags, more, entity) BLOCK;
+[%- MACRO tag_list(tags, entity, genres) BLOCK;
     tag_links = [];
     FOR tag=tags;
-        tag_links.push(link_tag(tag.tag));
+        IF genres == 1 AND tag.tag.is_genre_tag;
+            tag_links.push(link_tag(tag.tag));
+        ELSIF genres == 0 AND NOT tag.tag.is_genre_tag;
+            tag_links.push(link_tag(tag.tag));
+        END;
     END;
-    IF more;
-        tag_links.push(link_entity(entity, 'tags', l('more...')));
+    IF !tag_links.size;
+        lp('(none)', 'tag');
```

Comment:
Localization context should be `genre` for genre list.

```perl
        lp('(none'), genres ? 'genre' : 'tag');
```
Nitpicking: `is_genre_list` would be less confusing than `genres` with `tags` parameter.

Reply:
Can you give an example of what you mean by `is_genre_list` and how you'd use it?

Reply:
Exactly the same as `genres` in the current code. IIUC, it conditions whether the produced list should contain genres or other tags.

Reply:
Oh ok, just renaming the variable, not sure what I thought there.

For the localization issue, I think you have to do it like `is_genre_list ? lp('(none)', 'genre') : lp('(none)', 'tag');` otherwise we won't be able to extract the msgctxts.